### PR TITLE
fix(anrok): handle errors correctly, fix types

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1557/anrok-actions-returns-a-500-when-calling-actions

- Handle errors correctly 
Catch err was assumed to be AxiosError but it was a Javascript error

- Fix types for `notTaxedReason` can be missing from anrok response 
That was the original error

- Fix types for `validation_errors` it has never been an array

- Changed to async/await for clarity, there was no reason to mix both.

### CODE REVIEW

File: integration-templates/anrok/utils.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export function errorToObject(err: unknown) {
+    if ('response' in (err as any)) {
+        return (err as any).response.data;
+    }
```
Comment (TBonnin) [inline integration-templates/anrok/utils.ts:4]: would it make sense to use `isAxiosError`? here we assume that if there is `response` field we return `data` but there is no guarantuee

**Review by TBonnin — APPROVED** [review]
small comment about the AxiosError handling but looks good otherwise

File: integration-templates/anrok/utils.ts (hunk 1 above)
Comment (bodinsamuel) [reply integration-templates/anrok/utils.ts:4]: we don't allow import axios, maybe that's something we should change or expose in the sdk
(imo we will get rid of axios at some point so I wouldn't bother)

Comment (bodinsamuel) [reply integration-templates/anrok/utils.ts:4]: let me know if you prefer to have this fixed rn 👍🏻

Comment (TBonnin) [reply integration-templates/anrok/utils.ts:4]: don't spend too much time on that. it is fine for now

**Pull request merged**